File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int maxlen = 0;
        Set<Integer> numSet = new HashSet<>();

        for (int num : nums) {
            numSet.add(num);
        }

        for (int num : numSet) {
            if (!numSet.contains(num - 1)) {
                int currentNum = num;
                int length = 1;

                while (numSet.contains(currentNum + 1)) {
                    currentNum = currentNum + 1;
                    length += 1;
                }

                maxlen = Math.max(maxlen, length);
            }
        }

        return maxlen;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        longest_sequence = 0

        for num in num_set:
            if num - 1 not in num_set:
                current_sequence = 1
                current_num = num
                while current_num + 1 in num_set:
                    current_sequence += 1
                    current_num += 1
                longest_sequence = max(longest_sequence, current_sequence)
        
        return longest_sequence